src/Converter.cpp

```diff
@@ -237,357 +302,208 @@ void Converter::printBegin()
+    }  
+    msgpack_pack_uint16(&pk, v);    
+    return true;
+}
+bool Converter::visit_ (int32_t & v) { 
+    if (debug) {
+        printBegin();
+        std::cout << __FUNCTION__ << "(int32_t & v) got "<<  v  << std::endl; 
+    }   
+    msgpack_pack_int32(&pk, v);    
+    return true;
+}
+bool Converter::visit_ (uint32_t& v) { 
+    if (debug) {
+        printBegin();
+        std::cout << __FUNCTION__ << "(uint32_t& v) got "<<  v  << std::endl; 
+    }   
+    msgpack_pack_uint32(&pk, v);    
+    return true;
+}
+bool Converter::visit_ (int64_t & v) { 
+    if (debug) {
+        printBegin();
+        std::cout << __FUNCTION__ << "(int64_t & v) got "<<  v  << std::endl;
+    }    
+    msgpack_pack_int64(&pk, v);    
+    return true;
+}
+bool Converter::visit_ (uint64_t& v) { 
+    if (debug) {
+        printBegin();
+        std::cout << __FUNCTION__ << "(uint64_t& v) got "<<  v  << std::endl;  
+    }  
+    msgpack_pack_uint64(&pk, v);    
+    return true;
+}
+bool Converter::visit_ (float   & v) { 
+    if (debug) {
+        printBegin();
+        std::cout << __FUNCTION__ << "(float   & v) got "<<  v  << std::endl;
     }
-
-    return i;
+    msgpack_pack_float(&pk, v);
+    return true;
+}
+bool Converter::visit_ (double  & v) { 
+    if (debug) {
+        printBegin();
+        std::cout << __FUNCTION__ << "(double  & v) got "<<  v  << std::endl;  
+    }  
+    msgpack_pack_double(&pk, v);
+    
+    return true;
 }
 
-bool Converter::visit_(Typelib::Enum const& e)
+bool Converter::visit_(Typelib::Value const& v, Typelib::OpaqueType const& type)
 {
-    if(!part && verbose >= 3 + depth)
-    {
+    if (debug) {
         printBegin();
-        std::cout << "enum" << std::endl;
-    }
-
-    Typelib::Enum::integral_type intValue =
-        *reinterpret_cast<Typelib::Enum::integral_type*>(data + offset);
-    offset += sizeof(Typelib::Enum::integral_type);
-    try
-    {
-        std::string value = e.get(intValue);
-
-        msgpack_pack_str(&pk, value.size());
-        msgpack_pack_str_body(&pk, value.c_str(), value.size());
-    }
-    catch(Typelib::Enum::ValueNotFound& e)
-    {
-        msgpack_pack_int16(&pk, intValue);
-        std::cerr << "[pocolog2msgpack] Could not find a string representation "
-            << "for enum value " << intValue << "." << std::endl;
+        std::cout << __FUNCTION__ << "OpaqueType at "<< v.getData() << std::endl;
     }
-
     return true;
 }
 
-bool Converter::visit_(Typelib::Pointer const& type)
+bool Converter::visit_(Typelib::Value const& v, Typelib::Pointer const& type)
 {
-    if(!part && verbose >= 3 + depth)
-    {
+    if (debug) {
         printBegin();
-        std::cout << "pointer" << std::endl;
+        std::cout << __FUNCTION__ << "Pointer at "<< v.getData() <<   std::endl;
     }
-
+    
     depth += 1;
-    TypeVisitor::visit_(type);
+    auto retval = Typelib::ValueVisitor::visit_(v, type);
```

Comment:
This should be bool right? I think in this case we can write it down explicitly here. Otherwise we can cast it to bool to avoid confusing implicit casting.